File: Code_HackerEarth/perfect.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;
int
main(){
	vector <string> gc,pn;
	pn.push_back("66");
	pn.push_back("77");
	int N,start=0,end=2,counter=2,iter=1;
	cin>>N;
	while(counter < N){
		iter += 1;
		for(int i = start; i < end; i++){
			gc.push_back("6" + pn[i] + "6");
			counter++;
			if(counter == N)
				break;
		}
		for(int i = start; i < end; i++){
			gc.push_back("7" + pn[i] + "7");
			counter++;
			if(counter == N)
				break;
		}
		pn.clear();
		pn.resize(0);
		pn = gc;
		gc.clear();
		gc.resize(0);
		int size = end - start;
		N -= size;
		counter -= size;
		end = pow(2,iter);
	}
	cout<<pn[N-1];
}

File: Code_HackerEarth/perfect.py
num = int(input())
perfect_num = [66,77];
end = counter = 2
iter = 1

while counter < num:
	iter +=1;
	for i in range(0,end):
		perfect_num.append(int("6" + str(perfect_num[i]) + "6"))
		counter += 1

	for i in range(0,end):
		perfect_num.append(int("7" + str(perfect_num[i]) + "7"))
		counter += 1
		start = end;
		end = end+2**iter
print(perfect_num[num-1])
